File: eval/code_corpus/vulnerable/sqli.py
# INTENTIONALLY VULNERABLE — eval fixture, do not deploy
"""SQL injection — raw string interpolation of user input into a query.

Ground-truth finding: vuln_class=sqli on the cur.execute line below.
"""
import sqlite3


def get_user(conn: sqlite3.Connection, user_id: str):
    cur = conn.cursor()
    # VULN: user_id concatenated straight into SQL — classic SQLi sink.
    cur.execute("SELECT * FROM users WHERE id = '%s'" % user_id)
    return cur.fetchall()

File: eval/code_corpus/vulnerable/sqli.js
// # INTENTIONALLY VULNERABLE — eval fixture, do not deploy
// SQL injection — user input concatenated into a query string.
// Ground-truth finding: vuln_class=sqli on the db.query line below.

function getUser(db, userId) {
  // VULN: template-literal interpolation straight into SQL — SQLi sink.
  return db.query(`SELECT * FROM users WHERE id = '${userId}'`);
}

module.exports = { getUser };
